src/pytest_plugins/logging.py

```diff
@@ -0,0 +1,262 @@
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config: pytest.Config) -> None:
+    """
+    Initialize logging.
+
+    This goes to a lot of effort to ensure that a log file is created per worker
+    if xdist is used and that the timestamp used in the filename is the same across
+    main and all workers.
+    """
+    global file_handler
+
+    potential_subcommand = None
+    if len(sys.argv) > 1:
+        potential_subcommand = sys.argv[1]
+    log_stem = getattr(config, "workerinput", {}).get("log_stem") or get_log_stem(
+        sys.argv[0], potential_subcommand
+    )
+
+    worker_id = os.getenv("PYTEST_XDIST_WORKER", "main")
+    log_filename = f"{log_stem}-{worker_id}.log"
+    log_path = Path("logs")
+    log_path.mkdir(exist_ok=True)
+    config.option.eest_log_file_path = log_path / log_filename
+
+    formatter = UTCFormatter(fmt="%(asctime)s [%(levelname)s] %(name)s: %(message)s")
+
+    root_logger = logging.getLogger()
+    root_logger.setLevel(config.getoption("eest_log_level"))
+
+    file_handler = logging.FileHandler(config.option.eest_log_file_path, mode="w")
+    file_handler.setFormatter(formatter)
+    config._eest_file_handler = file_handler  # type: ignore[attr-defined]
+    root_logger.addHandler(file_handler)
+
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setFormatter(
+        ColorFormatter(fmt="%(asctime)s [%(levelname)s] %(name)s: %(message)s")
+    )
+    root_logger.addHandler(stream_handler)
+
+
+def pytest_report_header(config: pytest.Config) -> list[str]:
+    """Show the log file path in the test session header."""
+    if eest_log_file_path := config.option.eest_log_file_path:
+        return [f"Log file: {eest_log_file_path}"]
+    return []
+
+
+def pytest_terminal_summary(terminalreporter: TerminalReporter, exitstatus: int) -> None:
+    """Display the log file path in the terminal summary like the HTML report does."""
+    if terminalreporter.config.option.collectonly:
+        return
+    if eest_log_file_path := terminalreporter.config.option.eest_log_file_path:
+        terminalreporter.write_sep("-", f"Log file: {eest_log_file_path.resolve()}", yellow=True)
+
+
+def log_only_to_file(level: int, msg: str, *args, **kwargs) -> None:
+    """Log a message only to the file handler, bypassing stdout."""
+    if not file_handler:
+        return
+    handler: logging.Handler = file_handler  # type: ignore[attr-defined]
+    logger = logging.getLogger(__name__)
+    if not logger.isEnabledFor(level):
+        return
+    record: LogRecord = logger.makeRecord(
+        logger.name,
+        level,
+        fn=__file__,
+        lno=0,
+        msg=msg,
+        args=args,
+        exc_info=None,
+        func=None,
+        extra=None,
+    )
+    handler.handle(record)
+
+
+def pytest_runtest_logstart(nodeid: str, location: tuple[str, int, str]) -> None:
+    """Log test start to file."""
+    log_only_to_file(logging.INFO, f"ℹ️  - START TEST: {nodeid}")
```

Comment:
```suggestion
    log_only_to_file(logging.INFO, f"START TEST: {nodeid}")
```
As much as I love emojis I think we should not use them as they might not render correctly in all terminal environments!

Reply:
This only goes to file, where it's really helpful to have this visual guide, smol example:
```
2025-03-27 09:42:00.502+00:00 [INFO] pytest_plugins.logging: ℹ️  - START TEST: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Cancun-blockchain_test_from_state_test]-go-ethereum_default]
2025-03-27 09:42:00.531+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Starting client (go-ethereum_default)...
2025-03-27 09:42:01.409+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Client (go-ethereum_default) ready!
2025-03-27 09:42:01.411+00:00 [INFO] pytest_plugins.consume.hive_simulators.rlp.test_via_rlp: Calling getBlockByNumber to get genesis block...
2025-03-27 09:42:01.416+00:00 [INFO] pytest_plugins.consume.hive_simulators.rlp.test_via_rlp: Calling getBlockByNumber to get latest block...
2025-03-27 09:42:01.421+00:00 [INFO] pytest_plugins.logging: ✅ - PASSED in 0.01s: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Cancun-blockchain_test_from_state_test]-go-ethereum_default]
2025-03-27 09:42:01.421+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Stopping client (go-ethereum_default)...
2025-03-27 09:42:01.564+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Client (go-ethereum_default) stopped!
2025-03-27 09:42:01.569+00:00 [INFO] pytest_plugins.logging: ℹ️  - END TEST: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Cancun-blockchain_test_from_state_test]-go-ethereum_default]
2025-03-27 09:42:01.569+00:00 [INFO] pytest_plugins.logging: ℹ️  - START TEST: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Shanghai-blockchain_test_from_state_test]-go-ethereum_default]
2025-03-27 09:42:01.575+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Starting client (go-ethereum_default)...
2025-03-27 09:42:02.500+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Client (go-ethereum_default) ready!
2025-03-27 09:42:02.502+00:00 [INFO] pytest_plugins.consume.hive_simulators.rlp.test_via_rlp: Calling getBlockByNumber to get genesis block...
2025-03-27 09:42:02.507+00:00 [INFO] pytest_plugins.consume.hive_simulators.rlp.test_via_rlp: Calling getBlockByNumber to get latest block...
2025-03-27 09:42:02.512+00:00 [INFO] pytest_plugins.logging: ✅ - PASSED in 0.01s: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Shanghai-blockchain_test_from_state_test]-go-ethereum_default]
2025-03-27 09:42:02.513+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Stopping client (go-ethereum_default)...
2025-03-27 09:42:02.677+00:00 [INFO] pytest_plugins.consume.hive_simulators.conftest: Client (go-ethereum_default) stopped!
2025-03-27 09:42:02.688+00:00 [INFO] pytest_plugins.logging: ℹ️  - END TEST: src/pytest_plugins/consume/hive_simulators/rlp/test_via_rlp.py::test_via_rlp[tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid[fork_Shanghai-blockchain_test_from_state_test]-go-ethereum_default]
```

Reply:
aha! no stress then!